client/lxd_instances.go

```diff
@@ -1597,6 +1597,24 @@ func (r *ProtocolLXD) CreateInstanceFile(instanceName string, filePath string, a
 		req.Header.Set("X-LXD-write", args.WriteMode)
 	}
 
+	var modifyPerm []string
+
+	if args.UIDModifyExisting {
+		modifyPerm = append(modifyPerm, "uid")
+	}
+
+	if args.GIDModifyExisting {
+		modifyPerm = append(modifyPerm, "gid")
+	}
+
+	if args.ModeModifyExisting {
+		modifyPerm = append(modifyPerm, "mode")
+	}
+
+	if len(modifyPerm) != 0 && r.CheckExtension("instance_files_modify_permissions") == nil {
```

Comment:
@MggMuggins Looks like the issue with the permissions not updating after this change is just a typo. The API extension you've added is named `instances_files_modify_permissions`, but here when you check it, `instance` isn't pluralized so it's always skipping the header because it can't find it.

Reply:
Confused how the tests are passing though

Reply:
I think the unit test doesn't directly use the client, it instead supplies the headers directly rather than checking the API extension.

Reply:
Thanks Max. Fix & tests coming soon